app/services/katello/pulp3/repository_mirror.rb

```diff
@@ -24,11 +24,8 @@ def backend_object_name
       def refresh_entities
         href = remote_href
         if href
-          if remote_options[:url]&.start_with?('uln')
-            [api.remotes_uln_api.partial_update(href, remote_options)]
-          else
-            [api.remotes_api.partial_update(href, remote_options)]
-          end
+          # Do not consider remotes_uln_api, since the Katello server will never serve ULN!
```

Comment:
What does "...since the Katello server will never serve ULN" mean? Are ULN repos not supported on smart proxies?

Reply:
"ULN repo" just means an RPM repo with a specific type of authentication mechanism on it. Katello will sync the ULN repo to the Katello server, and then publish a normal Pulp RPM repo on the Katello server. This normal RPM repo is then synced to smart proxy the normal way. The sync from Katello server to smart proxy will never use a ULN URL. Therefore it will never use a ULN remote. We will never implement ULN authentication for the repos served in pulp_rpm/Katello.

Reply:
I suppose checking the URL that is synced to the proxy for ULN would not hurt, it's just that test will always come up: "not ULN".

Reply:
Would my comment be more clear with something like:
```suggestion
          # Do not consider remotes_uln_api, since the Katello server is not a ULN server. Even if the sync to Katello used ULN, the sync from Katello server to smart proxy will use a normal RPM remote!
```

Reply:
Thanks for the clarification, makes sense now.  I didn't realize that the ULN remote pretty much existed just for their special authentication. 

I think the updated comment would be helpful.

Reply:
I may have oversimplified a little bit, but overall the "special authentication" mental model has served me fine.

I updated the comment.